reader/src/main/java/org/jline/reader/impl/LineReaderImpl.java

```diff
@@ -586,7 +593,12 @@ public String readLine(String prompt, String rightPrompt, MaskingCallback maskin
 
             smallTerminalOffset = 0;
 
-            state = State.NORMAL;
+            if (isDumb && LF.equals(getLastBinding())) {
```

Comment:
This sounds wrong.  This is the preparation of the `readLine` call, so we should not discard bindings here.  If there's a cr+lf in a previous call, it should have been consumed completely.

Reply:
See https://github.com/jline/jline3/pull/874/files#r1365449433

Reply:
See my comment on the other PR. I don't think that is possible.

Reply:
Note that we are not discarding anything there. Here we just check if the previous `ACCEPT_LINE` was triggered  by an `LF`. The `getLastBinding` returns `LF` if that is the case. It doesn't consume the binding.